py/desispec/test/test_calibfinder.py

```diff
@@ -7,40 +7,74 @@
 import os
 import shutil
 from importlib import resources
-
-
+import tempfile
 
 from desispec.calibfinder import CalibFinder
 
+_standard_calib_dirs = ('DESI_SPECTRO_CALIB' in os.environ) and ('DESI_SPECTRO_DARK' in os.environ)
+
 class TestCalibFinder(unittest.TestCase):
     """Test desispec.calibfinder
     """
-    def tearDown(self):
-        pass
-        if os.path.isdir(self.calibdir) :
-            shutil.rmtree(self.calibdir)
-
-    def setUp(self):    
-        #- Create temporary calib directory
-        self.calibdir  = os.path.join(os.environ['HOME'], 'preproc_unit_test')
-        if not os.path.exists(self.calibdir): os.makedirs(self.calibdir)
-        #- Copy test calibration-data.yaml file 
-        specdir=os.path.join(self.calibdir,"spec/sp0")
-        if not os.path.isdir(specdir) :
-            os.makedirs(specdir)
+
+    @classmethod
+    def setUpClass(cls):
+
+        #- Cache original environment
+        cls.origenv = dict()
+        for key in ['DESI_SPECTRO_CALIB', 'DESI_SPECTRO_DARK']:
+            cls.origenv[key] = os.getenv(key, None)
+
+        #- Prepare alternate $DESI_SPECTRO_CALIB for testing
+        cls.calibdir = tempfile.mkdtemp()
+        specdir = os.path.join(cls.calibdir,"spec/sp0")
+        os.makedirs(specdir)
         for c in "brz" :
             shutil.copy(str(resources.files('desispec').joinpath(f'test/data/ql/{c}0.yaml')), os.path.join(specdir,f"{c}0.yaml"))
-        #- Set calibration environment variable    
-        os.environ["DESI_SPECTRO_CALIB"] = self.calibdir
-        
+
+    @classmethod
+    def tearDownClass(cls):
+        #- remove temporary calibration directory
```

Comment:
Here's a perfect example of running `shutil.rmtree()`. Just a reference to another comment.